contrib/internal/fasthttptrace/fasthttpheaderscarrier.go

```diff
@@ -0,0 +1,45 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package fasthttptrace
+
+import (
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/valyala/fasthttp"
+)
+
+// FastHTTPHeadersCarrier implements tracer.TextMapWriter and tracer.TextMapReader on top
+// of fasthttp's RequestHeader object, allowing it to be used as a span context carrier for
+// distributed tracing.
+type FastHTTPHeadersCarrier struct {
+	ReqHeader *fasthttp.RequestHeader
+}
+
+var _ tracer.TextMapWriter = (*FastHTTPHeadersCarrier)(nil)
+var _ tracer.TextMapReader = (*FastHTTPHeadersCarrier)(nil)
+
+// ForeachKey iterates over fasthttp request header keys and values
+func (f *FastHTTPHeadersCarrier) ForeachKey(handler func(key, val string) error) error {
+	keys := f.ReqHeader.PeekKeys()
+	for _, key := range keys {
+		sKey := string(key)
+		vs := f.ReqHeader.PeekAll(sKey)
+		for _, v := range vs {
+			if err := handler(sKey, string(v)); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Set adds the given value to request header for key. Key will be lowercased to match
+// the metadata implementation.
+func (f *FastHTTPHeadersCarrier) Set(key, val string) {
+	// f.ReqHeader.Set(k, val)
+	// MOTFF: "Set" overwrites any value at `k`. "Add" appends it. Just confirming we want to append, not overwrite
```

Comment:
See discussion here: https://github.com/DataDog/dd-trace-go/pull/2118#discussion_r1321562999
TLDR; 
The definition for TextMapCarrier's Set method is nested and very vague: "Set implements TextMapWriter."
...The definition for TextMapWriter's Set method states, "Set sets the given key/value pair."

grpc's [mdcarrier](https://github.com/DataDog/dd-trace-go/blob/main/contrib/google.golang.org/internal/grpcutil/mdcarrier.go#L32-L37), which fulfills TextMapCarrier, appends rather than overwrites: https://github.com/DataDog/dd-trace-go/blob/main/contrib/google.golang.org/internal/grpcutil/mdcarrier.go#L32-L37

It seems that we'd just want to overwrite data, but I want to make sure this gets properly reviewed.
**Do we want to append or set/overwrite data?**